full/src/test/java/apoc/load/LoadDirectoryTest.java

```diff
@@ -0,0 +1,111 @@
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + "TestCsv1.csv");
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + "TestCsv2.csv");
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + "TestCsv3.csv");
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + "TestXls1.xsl");
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + "TestJson1.json");
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + SUBFOLDER + File.separator + "TestSubfolder.json");
+        temporaryFolder.newFile(IMPORT_DIR + File.separator + SUBFOLDER + File.separator + "TestSubfolder.csv");
+    }
+
+    @Test
+    public void testWithNullUrlDir() {
+        apocConfig().setProperty(APOC_IMPORT_FILE_USE_NEO4J_CONFIG, true);
+        TestUtil.testFail(db, "CALL apoc.load.directory('*', null, {recursive: false}) YIELD value RETURN value", IllegalArgumentException.class);
+    }
+
+    @Test
+    public void testWithSubfolder() {
+        apocConfig().setProperty(APOC_IMPORT_FILE_USE_NEO4J_CONFIG, true);
+        testResult(db, "CALL apoc.load.directory('*', 'subfolder', {recursive: false}) YIELD value RETURN value", result -> {
+                    List<Map<String, Object>> rows = Iterators.asList(result.columnAs("value"));
+                    assertTrue(rows.contains(SUBFOLDER + File.separator + "TestSubfolder.json"));
+                    assertTrue(rows.contains(SUBFOLDER + File.separator + "TestSubfolder.csv"));
+                    assertEquals(2, rows.size());
+                }
+        );
+    }
+
+    @Test
+    public void testWithFileProtocol() {
+        apocConfig().setProperty(APOC_IMPORT_FILE_USE_NEO4J_CONFIG, false);
+        File rootTempFolder = temporaryFolder.getRoot();
+        String folderAsExternalUrl = "file://" + rootTempFolder;
+        testResult(db, "CALL apoc.load.directory('*', '" + folderAsExternalUrl + "', {recursive: false}) YIELD value RETURN value", result -> {
+                    List<Map<String, Object>> rows = Iterators.asList(result.columnAs("value"));
+                    assertTrue(rows.contains(rootTempFolder + File.separator + "Foo.csv"));
+                    assertTrue(rows.contains(rootTempFolder + File.separator + "Bar.csv"));
+                    assertTrue(rows.contains(rootTempFolder + File.separator + "Baz.xls"));
+                    assertEquals(3, rows.size());
+                }
+        );
+    }
+
+    @Test
+    public void testWithFilterAllRecursiveFalse() {
+        apocConfig().setProperty(APOC_IMPORT_FILE_USE_NEO4J_CONFIG, true);
+        testResult(db, "CALL apoc.load.directory('*', '', {recursive: false}) YIELD value RETURN value", result -> {
+                    List<Map<String, Object>> rows = Iterators.asList(result.columnAs("value"));
+                    assertTrue(rows.contains("TestCsv3.csv"));
+                    assertTrue(rows.contains("TestJson1.json"));
+                    assertEquals(5, rows.size());
```

Comment:
what about asserting the 3 other file names explicitly?

Reply:
certainly, I added them